src/cook-web/src/store/useShifu.tsx

```
@@ -268,49 +256,35 @@ export const ShifuProvider: React.FC<{ children: ReactNode }> = ({
+    const types = Object.fromEntries(
+      list.map((block: Block) => [block.bid, block.type])
+    )
     setBlockUITypes(types)
   }
 
   const initBlockUIProperties = async (list: Block[]) => {
     const properties = list.reduce((prev: any, cur: Block) => {
       return {
         ...prev,
-        [cur.properties.block_id]: cur.properties.block_ui.properties
+        [cur.bid]: cur.properties
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)